src/coder/lg_server.py

```diff
@@ -1,14 +1,42 @@
 """Coder graph for LangGraph Server."""
 
-from coder.graph import graph_builder
+import os
+from typing import Union
+
+from langchain_community.utilities.github import GitHubAPIWrapper
+
+from coder.graph import coder_change_request_config, coder_new_pr_config
 from coder.mocks import MockGithubApi
 from coder.tools import get_github_tools
 
-# graph for LangGraph Server
-# TODO: maybe we can setup this to use mock/real github api depending on env/config
-mock_api = MockGithubApi()
-github_tools = get_github_tools(mock_api)
-graph = graph_builder(github_tools).compile()
-graph.name = "Coder"
 
-__all__ = ["graph"]
+def get_github_source() -> Union[GitHubAPIWrapper, MockGithubApi]:
+    """Get either a real GitHub API wrapper or a mock based on environment variables.
+
+    Required environment variables for real GitHub API:
+    - GITHUB_APP_ID
+    - GITHUB_APP_PRIVATE_KEY
+    - GITHUB_REPOSITORY
+    """
+    required_vars = ["GITHUB_APP_ID", "GITHUB_APP_PRIVATE_KEY", "GITHUB_REPOSITORY"]
+
+    if all(os.getenv(var) for var in required_vars):
```

Comment:
Could be confusing if one of these isn't provided and we fall back to `MockGithubApi`, maybe a log statement or error in that case would help.